From 3a87e1e74c171ebfb826f053f54f9f323074153e Mon Sep 17 00:00:00 2001
From: "Ch1llb0/skin.osmc" <36623079+Ch1llb0@users.noreply.github.com>
Date: Mon, 17 Dec 2018 18:32:38 +0100
Subject: [PATCH] Bug fixes #2

Defaults.xml:
- change invalid color to corresponding variable

script-skinshortcuts.xml:
- fix focus text color
- fix disabled items' text color

Settings.Category.xml:
- change disabled color to corresponding variable
- change invalid color to corresponding variable

SettingsProfile.xml:
- change disabled color to corresponding variable
- change invalid color to corresponding variable

Variables.xml:
- change selected color for OSMC Blue and OSMC Light blue
- chane skinshortcuts-indicator variable for skinshortcuts disabled items' text color fix
---
 16x9/Defaults.xml             |  4 ++--
 16x9/SettingsCategory.xml     | 14 +++++++-------
 16x9/SettingsProfile.xml      |  4 ++--
 16x9/Variables.xml            | 24 ++++++++++++------------
 16x9/script-skinshortcuts.xml | 22 +++++-----------------
 5 files changed, 28 insertions(+), 40 deletions(-)

diff --git a/16x9/Defaults.xml b/16x9/Defaults.xml
index 90d996d..4480b71 100755
--- a/16x9/Defaults.xml
+++ b/16x9/Defaults.xml
@@ -33,7 +33,7 @@
 		<textoffsetx>0</textoffsetx>
 		<textcolor>$VAR[TextColor2]</textcolor>
 		<focusedcolor>$VAR[TextColor1]</focusedcolor>
-		<invalidcolor>$VAR[DisabledColor]</invalidcolor>
+		<invalidcolor>$VAR[InvalidColor]</invalidcolor>
 		<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 		<texturefocus colordiffuse="$VAR[buttonfocus]">common/focus.png</texturefocus>
 		<texturenofocus></texturenofocus>
@@ -176,7 +176,7 @@
 		<textcolor>$VAR[TextColor2]</textcolor>
 		<focusedcolor>$VAR[TextColor1]</focusedcolor>
 		<disabledcolor>$VAR[DisabledColor]</disabledcolor>
-		<invalidcolor>$VAR[DisabledColor]</invalidcolor>
+		<invalidcolor>$VAR[InvalidColor]</invalidcolor>
 		<texturefocus colordiffuse="$VAR[buttonfocus]">common/focus.png</texturefocus>
 		<pulseonselect>no</pulseonselect>
 	</default>
diff --git a/16x9/SettingsCategory.xml b/16x9/SettingsCategory.xml
index df4d06c..6386e58 100755
--- a/16x9/SettingsCategory.xml
+++ b/16x9/SettingsCategory.xml
@@ -217,8 +217,8 @@
 			<onclick>SetFocus(5)</onclick>
 			<textcolor>$VAR[TextColor2]</textcolor>
 			<focusedcolor>$VAR[TextColor1]</focusedcolor>
-			<invalidcolor>$VAR[TextColor4]</invalidcolor>
-			<disabledcolor>$VAR[TextColor4]</disabledcolor>
+			<invalidcolor>$VAR[InvalidColor]</invalidcolor>
+			<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 			<animation effect="fade" start="100" end="50" time="200" tween="cubic" easing="out" reversible="false" condition="ControlGroup(5).HasFocus | Control.HasFocus(20)">Conditional</animation>
 		</control>
 
@@ -230,8 +230,8 @@
 			<textwidth>900</textwidth>
 			<textcolor>$VAR[TextColor2]</textcolor>
 			<focusedcolor>$VAR[TextColor1]</focusedcolor>
-			<invalidcolor>$VAR[TextColor4]</invalidcolor>
-			<disabledcolor>$VAR[TextColor4]</disabledcolor>
+			<invalidcolor>$VAR[InvalidColor]</invalidcolor>
+			<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 		</control>
 
 		<!-- Default radiobutton -->
@@ -242,7 +242,7 @@
 			<textwidth>900</textwidth>
 			<textcolor>$VAR[TextColor2]</textcolor>
 			<focusedcolor>$VAR[TextColor1]</focusedcolor>
-			<disabledcolor>$VAR[TextColor4]</disabledcolor>
+			<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 		</control>
 
 		<!-- Default spincontrolex -->
@@ -253,7 +253,7 @@
 			<textwidth>900</textwidth>
 			<textcolor>$VAR[TextColor2]</textcolor>
 			<focusedcolor>$VAR[TextColor1]</focusedcolor>
-			<disabledcolor>$VAR[TextColor4]</disabledcolor>
+			<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 		</control>
 
 		<!-- Default sliderex -->
@@ -264,7 +264,7 @@
 			<textwidth>900</textwidth>
 			<textcolor>$VAR[TextColor2]</textcolor>
 			<focusedcolor>$VAR[TextColor1]</focusedcolor>
-			<disabledcolor>$VAR[TextColor4]</disabledcolor>
+			<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 		</control>
 
 		<!-- Default image -->
diff --git a/16x9/SettingsProfile.xml b/16x9/SettingsProfile.xml
index be8a5db..6eb84fe 100755
--- a/16x9/SettingsProfile.xml
+++ b/16x9/SettingsProfile.xml
@@ -47,8 +47,8 @@
 					<font>Font33</font>
 					<textcolor>$VAR[TextColor2]</textcolor>
 					<focusedcolor>$VAR[TextColor1]</focusedcolor>
-					<invalidcolor>$VAR[TextColor4]</invalidcolor>
-					<disabledcolor>$VAR[TextColor4]</disabledcolor>
+					<invalidcolor>$VAR[InvalidColor]</invalidcolor>
+					<disabledcolor>$VAR[DisabledColor]</disabledcolor>
 					<label>33084</label>
 					<enable>!System.HasLoginScreen</enable>
 				</control>
diff --git a/16x9/Variables.xml b/16x9/Variables.xml
index e62c533..56af873 100755
--- a/16x9/Variables.xml
+++ b/16x9/Variables.xml
@@ -145,8 +145,8 @@
 	
 	<!-- Selected color -->
 	<variable name="SelectedColor">
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue)">FFBC9A24</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue)">FFC76100</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue)">FF29B6EB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue)">FF00D7C7</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Lime green)">FF72FF88</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Sunset Yellow)">FFFFFC43</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.selected),None)]] + String.IsEqual(Skin.String(DefaultColorSet),Orange)">FFFF8536</value>
@@ -327,10 +327,10 @@
 	
 	<!-- Background color -->
 	<variable name="BackgroundColor">
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),High)">CC2435bc</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Medium)">A62435bc</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Low)">802435bc</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Highest)">F22435bc</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),High)">CC293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Medium)">A6293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Low)">80293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Highest)">F2293FEB</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),High)">CC009CC7</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Medium)">A6009CC7</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.background),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultBackgroundOpacity),Low)">80009CC7</value>
@@ -377,10 +377,10 @@
 	
 	<!-- Overlay color -->
 	<variable name="OverlayColor">
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),High)">CC1c2ca5</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Medium)">991c2ca5</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Low)">661c2ca5</value>
-		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Highest)">FF1c2ca5</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),High)">CC293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Medium)">99293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Low)">66293FEB</value>
+		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Highest)">FF293FEB</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),High)">CC008AB0</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Medium)">99008AB0</value>
 		<value condition="[String.IsEqual(Skin.String(DefaultColors),yes) | [String.IsEqual(Skin.String(DefaultColors),no) + String.IsEqual(Skin.String(color.overlay),None)]] + String.IsEqual(Skin.String(DefaultColorSet),OSMC Light Blue) + String.IsEqual(Skin.String(DefaultOverlayOpacity),Low)">66008AB0</value>
@@ -1136,8 +1136,8 @@
 	</variable>
 
 	<variable name="skinshortcuts-disableindicator">
-		<value condition="String.IsEqual(ListItem.Property(skinshortcuts-disabled),True)">[COLOR darkred]Ø›/COLOR] </value>
-		<value />
+		<value condition="String.IsEqual(ListItem.Property(skinshortcuts-disabled),True)">[COLOR darkred]$VAR[skinshortcuts-itemlabel][/COLOR]</value>
+		<value>$VAR[skinshortcuts-itemlabel]</value>
 	</variable>
 
 	<variable name="skinshortcuts-size">
diff --git a/16x9/script-skinshortcuts.xml b/16x9/script-skinshortcuts.xml
index 785f451..33dcdca 100755
--- a/16x9/script-skinshortcuts.xml
+++ b/16x9/script-skinshortcuts.xml
@@ -67,7 +67,7 @@
 						<width>260</width>
 						<height>66</height>
 						<textcolor>$VAR[TextColor2]</textcolor>
-						<label>$VAR[skinshortcuts-disableindicator]$VAR[skinshortcuts-itemlabel]</label>
+						<label>$VAR[skinshortcuts-disableindicator]</label>
 					</control>
 				</itemlayout>
 
@@ -78,33 +78,21 @@
 						<height>66</height>
 						<scroll>true</scroll>
 						<textcolor>$VAR[TextColor1]</textcolor>
-						<label>$VAR[skinshortcuts-disableindicator]$VAR[skinshortcuts-itemlabel]</label>
-						<animation effect="fade" start="100" end="75" time="0" condition="!Control.HasFocus(211)">Conditional</animation>
-						<visible>Control.HasFocus(303) | Control.HasFocus(304)</visible>
-					</control>
-					<control type="label">
-						<left>25</left>
-						<width>260</width>
-						<height>66</height>
-						<scroll>true</scroll>
-						<textcolor>$VAR[TextColor2]</textcolor>
-						<label>$VAR[skinshortcuts-disableindicator]$VAR[skinshortcuts-itemlabel]</label>
-						<animation effect="fade" start="100" end="75" time="0" condition="!Control.HasFocus(211)">Conditional</animation>
-						<visible>![Control.HasFocus(303) | Control.HasFocus(304)]</visible>
+						<label>$VAR[skinshortcuts-disableindicator]</label>
 					</control>
 					<control type="image">
 						<left>25</left>
 						<width>335</width>
 						<height>66</height>
 						<texture colordiffuse="$VAR[TextColor1]">$VAR[focus66]</texture>
-						<visible>Control.HasFocus(303) | Control.HasFocus(304)</visible>
+						<visible>String.IsEqual(ListItem.Property(skinshortcuts-disabled),False)</visible>
 					</control>
 					<control type="image">
 						<left>25</left>
 						<width>335</width>
 						<height>66</height>
-						<texture colordiffuse="$VAR[TextColor2]">$VAR[focus66]</texture>
-						<visible>![Control.HasFocus(303) | Control.HasFocus(304)]</visible>
+						<texture colordiffuse="darkred">$VAR[focus66]</texture>
+						<visible>String.IsEqual(ListItem.Property(skinshortcuts-disabled),True)</visible>
 					</control>
 					<control type="image">
 						<left>290</left>
